README.md

```diff
@@ -1,9 +1,113 @@
 
-The project is under development. Stay tuned.
\ No newline at end of file
+We are always trying to improve it. Feel free to start a conversation or open an issue. Pull requests are welcome as well!
+
+# Table of contents
+
+1. [Core principles](#core-principles)
+2. [Easiest way to use](#the-easiest-way-to-use-the-library)
+3. [Importing from HTML](#importing-from-the-html)
+4. [NodeJS and JavaScript bundlers](#working-with-nodejs--javascript-bundlers)
+5. [SimpleDocument (experimental)](#simpledocument-experimental-for-nodejs-only)
+6. [All functions and methods](#all-functions-and-methods-available)
+
+## Core principles
+
+The library expects **tabular data** stored in CSV files or **arrays of objects** stored in JSON files. It works best when the data is tidy:
+
+1. Every column (or key) is a variable
+
+2. Every row (or item) is an observation
+
+3. Every cell (or value) is a single value
+
+## The easiest way to use the library
+
+If you don't want to install anything, a great platform is Observable. Here's how to use the library inside an [Observable's notebook](https://observablehq.com/@nshiab/simple-data-analysis).
+
+![An Observable notebook using simple-data-analysis](./assets/observable.png)
+
+## Importing from the HTML
+
+If you want to add the library directly to your webpage, you can use the UMD minified bundle. Call **sda** to have access to the functions.
+
+```js
+<script src="https://cdn.jsdelivr.net/npm/simple-data-analysis@latest"></script>
+
+<script>
+
+const someData = [
+    {firstName: "Nael", lastName: "Shiab", job: "Computational journalist"},
+    {firstName: "Isabelle", lastName: "Bouchard", job: "Data scientist"}
+]
+
+const simpleData = sda.createSimpleData(someData)
+
+</script>
+```
+## Working with NodeJS / JavaScript Bundlers
+
+First, make sure that your NodeJS version is 16 or higher. To check, write ```node``` in your terminal and press Enter.
+
+You should see something like this.
+![A terminal showing the NodeJS version](/assets/nodeJSVersion.png)
+
+If the version is less than 16, update [NodeJS](https://nodejs.org/en/) with the latest LTS (long-term support) version.
+
+To install the library with npm, type this command in your terminal:
+```
+npm i simple-data-analysis
+```
+
+Once installed, you can import the functions as needed. If you use a bundler (Webpack, Rollup, Parcel or others), importing only the required functions will make your final project lighter.
+
+**/!\ This is how you should import the functions if you plan to publish your project on the web. /!\\**
+```js
+import {createSimpleData} from "simple-data-analysis"
+
+const simpleData = createSimpleData(someData)
+```
+
+But you can also import everything if you wish.
+```js
+import * as sda from "simple-data-analysis"
+
+const simpleData = sda.createSimpleData(someData)
+```
+
+## SimpleDocument (experimental, for NodeJS only)
```

Comment:
I would have a similar section for `SimpleData`. You could show how you can easily chain! 🎉